src/main/java/net/glowstone/entity/GlowEntity.java

```diff
@@ -616,37 +616,47 @@ public void pulse() {
             }
         }
         GlowBlock block = world.getBlockAt(location);
+        // Nether portal handling
         if (block.getType().equals(Material.PORTAL)
             && BlockPortal.getBoundingBox(block).intersects(boundingBox)) {
             if (!isInPortal) {
+                //Entity just entered the Portal, fire EnterEvent
                 EventFactory.getInstance().callEvent(
                     new EntityPortalEnterEvent(this, block.getLocation()));
+                // prevent EnterEvent being fired again
                 isInPortal = true;
             }
+            // only try porting if the nether is enabled
             if (server.getAllowNether() && portalCooldown <= 0) {
-                GlowWorld w = getWorld().getEnvironment().equals(Environment.NETHER)
+                // determine destination world
+                GlowWorld destination = getWorld().getEnvironment().equals(Environment.NETHER)
```

Comment:
It might be necessary to ensure that there is such a world, even if the `allowNether` configuration is enabled.

Reply:
This logic would allow portals from the End to the Nether without cheats, which aren't possible in vanilla.

Reply:
Should we just add a check for the type of world or something else?

Reply:
Perhaps look into adding a method to GlowWorld along the lines of:

```
boolean isPortalDestinationValid(BlockType type, GlowWorld destination)
```

This way, we aren't hardcoding Overworld to Nether, or Overworld to End, allowing for more extensibility later on (including for plugins).

Reply:
Applied the suggestions. Thanks for the input